dev/NavigationView/NavigationViewItemPresenter.cpp

```diff
@@ -54,4 +66,24 @@ NavigationViewItem* NavigationViewItemPresenter::GetNavigationViewItem()
         navigationViewItem = winrt::get_self<NavigationViewItem>(item);
     }
     return navigationViewItem;
-}
\ No newline at end of file
+}
+
+void NavigationViewItemPresenter::UpdateIndentations()
+{
+    winrt::IControlProtected controlProtected = *this;
+    auto presenterContentGrid = GetTemplateChildT<winrt::Grid>(c_navigationViewItemPresenterContentGridName, controlProtected);
+    if (presenterContentGrid)
+    {
+        auto leftIndentation = s_indentation * m_depth;
+        auto thickness = winrt::ThicknessHelper::FromLengths(leftIndentation, 0, 0, 0);
+        presenterContentGrid.Margin(thickness);
```

Comment:
>Margin [](start = 29, length = 6)

For new features, we don't prefer to change UI from code. 
TemplateSettings is a good choice, for example
<Grid x:Name="TogglePaneTopPadding"
                                  Height="{Binding RelativeSource={RelativeSource TemplatedParent}, Path=TemplateSettings.TopPadding}"/>